jbpm-console-ng-executor-service/jbpm-console-ng-executor-service-client/src/test/java/org/jbpm/console/ng/es/client/editors/requestlist/dataset/DataSetRequestListPresenterTest.java

```diff
@@ -0,0 +1,123 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jbpm.console.ng.es.client.editors.requestlist.dataset;
+
+
+import javax.enterprise.event.Event;
+import javax.inject.Inject;
+
+import com.google.gwt.view.client.Range;
+import com.google.gwtmockito.GwtMockitoTestRunner;
+import org.dashbuilder.dataset.sort.SortOrder;
+import org.jbpm.console.ng.df.client.filter.FilterSettings;
+import org.jbpm.console.ng.df.client.filter.FilterSettingsBuilderHelper;
+import org.jbpm.console.ng.df.client.list.base.DataSetQueryHelper;
+import org.jbpm.console.ng.es.model.RequestSummary;
+import org.jbpm.console.ng.es.model.events.RequestChangedEvent;
+import org.jbpm.console.ng.es.service.ExecutorServiceEntryPoint;
+import org.jbpm.console.ng.gc.client.experimental.grid.base.ExtendedPagedTable;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.uberfire.client.workbench.widgets.common.ErrorPopupPresenter;
+import org.uberfire.mocks.CallerMock;
+
+import static org.dashbuilder.dataset.sort.SortOrder.*;
+import static org.mockito.Mockito.*;
+
+@RunWith(GwtMockitoTestRunner.class)
+public class DataSetRequestListPresenterTest {
+    private static final Long TASK_ID = 1L;
+    private static final String USR_ID = "admin";
+
+    private CallerMock<ExecutorServiceEntryPoint> callerMockExecutorService;
+
+    @Mock
+    private ExecutorServiceEntryPoint executorServiceMock;
+
+
+    @Mock
+    private DataSetRequestListViewImpl viewMock;
+
+    @Mock
+    DataSetQueryHelper dataSetQueryHelper;
+
+    @Mock
+    private ExtendedPagedTable<RequestSummary> extendedPagedTable;
+
+    @Mock
+    private ErrorPopupPresenter errorPopup;
+
+    private FilterSettings filterSettings;
+
+    //Thing under test
+    private DataSetRequestListPresenter presenter;
+
+    @Before
+    public void setupMocks() {
+        //Mock that actually calls the callbacks
+        callerMockExecutorService = new CallerMock<ExecutorServiceEntryPoint>(executorServiceMock);
+
+        filterSettings= createTableSettingsPrototype();
+
+        when(viewMock.getListGrid()).thenReturn(extendedPagedTable);
+        when(extendedPagedTable.getPageSize()).thenReturn(10);
+        when(extendedPagedTable.getColumnSortList()).thenReturn(null);
+        when(dataSetQueryHelper.getCurrentTableSettings()).thenReturn(filterSettings);
+
+        //dataSetQueryHelper.setCurrentTableSettings(createTableSettingsPrototype());
+        presenter = new DataSetRequestListPresenter(viewMock,
+                callerMockExecutorService,dataSetQueryHelper);
+    }
+
+    @Test
+    public void getDataTest() {
+
+        presenter.setAddingDefaultFilters(false);
+        presenter.getData(new Range(0, 5));
+
+
+        verify(dataSetQueryHelper).setLastSortOrder(SortOrder.ASCENDING);
+        verify(viewMock).hideBusyIndicator();
+
+    }
+
```

Comment:
The important things to test here is that getMenus calls your new setupRefreshButton which is the fix submitted in this PR.